### PR TITLE
some refactoring for cli and api location tests 

### PR DESCRIPTION
**Pull request by pondrejk**
merged and removed tests based on https://mojo.redhat.com/groups/satellite6qe/blog/2019/06/04/making-robottelo-slimmer

Test results:
```
pytest tests/foreman/cli/test_location.py 
================================================================= test session starts =================================================================
platform linux -- Python 3.7.3, pytest-4.0.2, py-1.7.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pondrejk/Documents/robottelo, inifile:
plugins: xdist-1.25.0, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.1
collecting ... 2019-06-10 09:59:59 - conftest - DEBUG - BZ deselect is disabled in settings

collected 8 items                                                                                                                                     
tests/foreman/cli/test_location.py ........

8 passed, 2 warnings in 227.70 seconds
```
```pytest tests/foreman/api/test_location.py 
====================================================== test session starts ======================================================
platform linux -- Python 3.7.3, pytest-4.0.2, py-1.7.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pondrejk/Documents/robottelo, inifile:
plugins: xdist-1.25.0, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.1
collecting ... 2019-06-10 14:13:15 - conftest - DEBUG - BZ deselect is disabled in settings

collected 11 items                                                                                                              

tests/foreman/api/test_location.py ...........

11 passed, 2 warnings, 1 error in 177.51 seconds
```

For comparison, test result before this change
CLI: 1 failed, 45 passed, 8 warnings in 1406.91 seconds
API: 2 failed, 33 passed, 2 warnings in 477.85 seconds
So this change reduces the execution time to around 20% for combined api and cli test.

Brief rundown of changes:
rundown of changes:
- create tests with various imputs kept in the api
- create tests with parameters merged in the cli, reduced in api
- cli negative tests merged where appropriate
- cli add and remove parameter and capsule tests merged
- docker compute resource tests removed as docker cr no longer present
- create and update api tests merged

### CODE REVIEW

**Comment by jhutar** [comment]
So many lines removed - @pondrejk for president!
ACK :-)

File: tests/foreman/api/test_location.py [hunk 1]
```diff
@@ -66,6 +64,19 @@ def _make_proxy(self, options=None):
         self.addCleanup(capsule_cleanup, proxy['id'])
         return proxy
 
+    @classmethod
+    def setUpClass(cls):
```
Comment (mirzal) [inline tests/foreman/api/test_location.py]: From what I see, we usually also do something like:
```
super(LocationTestCase, cls).setUpClass()
```

Which will call `setUpClass` method of `APITestCase`.

File: tests/foreman/cli/test_location.py [hunk 2]
```diff
@@ -76,296 +46,89 @@ def _make_proxy(self, options=None):
...
-    @tier1
-    def test_positive_create_with_name(self):
-        """Try to create location using different value types as a name
-
-        :id: 76a90b92-296c-4b5a-9c81-183ff71937e2
-
-        :expectedresults: Location is created successfully and has proper name
-
+    @classmethod
+    def setUpClass(cls):
```
Comment (mirzal) [inline tests/foreman/cli/test_location.py]: The same comment about `super()` applies here.

File: tests/foreman/api/test_location.py (hunk 1 above)
Comment (pondrejk) [reply tests/foreman/api/test_location.py]: great chatch, updated

**Pull request merged**